crypto/des/fcrypt.c

```diff
@@ -70,9 +70,12 @@ char *DES_crypt(const char *buf, const char *salt)
     char e_buf[32 + 1];         /* replace 32 by 8 ? */
     char *ret;
 
+    if (sizeof(salt) < 2) return NULL;
```

Comment:
right; there's no way to test that the slt[] array is two bytes.  that's C for ya :)